ir/base.def

```diff
@@ -183,26 +183,30 @@ class Path {
     validate { BUG_CHECK(!name.name.isNullOrEmpty(), "Empty path"); }
 }
 
+/// Handy class used in several NamedMaps
+class NamedExpression : Declaration {
+    Expression expression;
+}
+
 /// Annotations are used to provide additional information to the compiler
 /// Most P4 entities can be optionally annotated
 class Annotation {
     ID name;
     /// Annotations that are simple expressions
     inline Vector<Expression> expr;
     /// Annotations described as key-value pairs
-    // TODO: this should be a map from ID not from cstring
-    inline NameMap<Expression> kv;
+    inline IndexedVector<NamedExpression> kv;
```

Comment:
why IndexedVector instead of NameMap?  You can use `NameMap<NamedExpression, ordered_map>` to specify an ordered map as the underlying map type (to preserve order semantics).

Reply:
These two are pretty much equivalent in functionality; they have slightly different APIs. For example, the IndexedVector only works with IDeclaration objects, and cannot support multisets of keys. I can convert to a NameMap if you prefer.